src/sentry/snuba/metrics/fields/snql.py

```diff
@@ -0,0 +1,72 @@
+from snuba_sdk import Column, Function
+
+from sentry.sentry_metrics.utils import resolve_weak
+
+
+def __counter_sum_aggregation_on_session_status_factory(session_status, metric_ids, alias=None):
+    return Function(
+        "sumMergeIf",
+        [
+            Column("value"),
+            Function(
+                "and",
+                [
+                    Function(
+                        "equals",
+                        [
+                            Column(f"tags[{resolve_weak('session.status')}]"),
+                            resolve_weak(session_status),
+                        ],
+                    ),
+                    Function("in", [Column("metric_id"), list(metric_ids)]),
+                ],
+            ),
+        ],
+        alias,
+    )
+
+
+def init_sessions(metric_ids, alias=None):
+    return __counter_sum_aggregation_on_session_status_factory(
+        session_status="init", metric_ids=metric_ids, alias=alias
+    )
+
+
+def crashed_sessions(metric_ids, alias=None):
+    return __counter_sum_aggregation_on_session_status_factory(
+        session_status="crashed", metric_ids=metric_ids, alias=alias
+    )
+
+
+def errored_preaggr_sessions(metric_ids, alias=None):
+    return __counter_sum_aggregation_on_session_status_factory(
+        session_status="errored_preaggr", metric_ids=metric_ids, alias=alias
+    )
+
+
+def sessions_errored_set(metric_ids, alias=None):
+    return Function(
+        "uniqCombined64MergeIf",
+        [
+            Column("value"),
+            Function(
+                "in",
+                [
+                    Column("metric_id"),
+                    list(metric_ids),
+                ],
+            ),
+        ],
+        alias,
+    )
+
+
+def percentage(arg1_snql, arg2_snql, metric_ids, alias=None):
+    return Function(
+        "multiply",
+        [
+            100,
+            Function("minus", [1, Function("divide", [arg1_snql, arg2_snql])]),
```

Comment:
What does snuba return for divide by zero?

Reply:
Not possible.. if there are no sessions, you just get no rows back so crash free rate is None